### ISSUE TITLE
Confusion about the Code Implementation

### ISSUE THREAD

**Issue by HenryLau7** [Ma-Lab-Berkeley/CRATE#18]
Thanks for your inspiring work!! Several confusion about the code implementation.

### 1.1 
In the setting, matrix $U$ stands for the orthonormal bases of the Gaussian space. But in the implement, it's a `nn.Linear()` implementation. 
### 1.2
Besides, in the MSSA equation,

$$
MSSA(Z \mid U_{[K]}) = \beta \begin{bmatrix}
U_1, \dots, U_K
\end{bmatrix} \begin{bmatrix}
SSA(Z \mid U_1) \\
\vdots \\
SSA(Z \mid U_K)
\end{bmatrix}.
$$

There is a $U^* SSL$ product, where $U$ is supposed to be the same as the $U$ in SSA. But there is another `to_out = nn.Linear` implementation. I think it destroys the theoretical property.

## 2. ISTA
In ISTA module, there is a dictionary $D$, where $D^*D\approx I$. However the optimization didn't include this constraint, and the implementation didn't consider this constraint. I think it may result in the property didn't hold.

Be appreciate for the reply! Thanks again!

**Comment by DruvPai** [comment]
These are great questions! In the response, I refer to [our new long-form work](https://arxiv.org/abs/2311.13110) as "the paper". It contains a superset of the results from our original NeurIPS paper on CRATE as well as two other works (accepted at CPAL 2024, ICLR 2024).

- Regarding not enforcing orthogonal optimization: We show in the paper that the learned $U$'s and $D$'s have incoherent columns, meaning that the orthogonality is (approximately) maintained even without needing to explicitly encode it into our optimization algorithm.

- Regarding the output matrix in the MSSA, we discuss this in the paper's appendix. The short answer is that the MSSA block, even in the theory, is just a means to an end: it is an _approximate_ implementation of an operator which minimizes the sparse rate reduction of its input. Since the MSSA block in the paper is approximately a gradient of  the compression term $R^{c}$, we have reason to believe that the MSSA block will optimize the sparse rate reduction. If we make a reasonably small modification to this operator (say to get the implemented MSSA block) which is a lot easier to train and scale, then we just have to show this new operator also acts to minimize the sparse rate reduction, which we do in the paper. (A comparison about the training and performance is included in the appendix of the paper.)
